include/CL/sycl/vec/detail/vec.hpp

```diff
@@ -115,7 +117,13 @@ class vec : public detail::small_array<DataType,
 
   /// Return the number of bytes
   auto get_size() const {
-    return NumElements * sizeof(DataType);
+    // Special case, aligning a size 3 vec to 4 as specified in
+    // SYCL 1.2.1 Section 4.10.2.6 should increase its size to
+    // the same as a vec4
+    if constexpr (NumElements == 3)
+      return 4 * sizeof(DataType);
+    else
+      return NumElements * sizeof(DataType);
```

Comment:
Perhaps you can just call the alignment trait instead of repeating the same code?